### PR TITLE
[ui] Add a chip to indicate pro features

### LINKED ISSUE #3181: Authorization paid plan error message
### Steps to reproduce

Getting an error message when on a free plan

Screenshot 2024-02-08 at 11 41 14

### Current behavior

_No response_

### Expected behavior

Options:
1. No _Authorization_ section visible. 
2. A "pro" badge after Authorization title and a help icon that links to relevant documentation.

### Context

_No response_

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: authentication free

### PR DESCRIPTION
**Pull request by bharatkashyap**
- Closes #3181

https://github.com/mui/mui-toolpad/assets/19550456/9b547809-80b9-4df0-80fb-e0bf7b61d425

- Also makes fixes to the `<UpgradeAlert />` to change the nature of the alerts to `info` instead of `error`, which was jarring

Screenshot 2024-04-05 at 1 06 17 PM

- Also some minor fixes on the typography of the `<PageOptionsPanel />`':
Screenshot 2024-04-05 at 1 09 15 PM

- Also some updates to documentation copy to better reflect the paid nature of the gated features

```
- "only available for free for a limited time"
+ "is a paid feature"
```

### CODE REVIEW

**Comment by oliviertassinari** [comment]
I think we should remove the blue icon, this is meant to represent MUI X Pro != Toolpad Commercial plan.

**Review by Janpot — APPROVED** [review]
I'm not convinced this will scale as the amount pro features grows, but that will be a good problem to have. Until then I'm ok with this solution 👍

File: packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx [hunk 1]
```diff
@@ -122,7 +122,38 @@ export default function PageOptionsPanel() {
         </Tooltip>
       </div>
       <div>
-        <Typography variant="body2">Authorization:</Typography>
+        <Typography variant="overline">
+          Authorization{' '}
+          {!isPaidPlan ? (
+            <Tooltip
+              title={
+                <Typography variant="inherit">
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx:130]:
```suggestion
```

File: packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx [hunk 2]
```diff
@@ -122,7 +122,38 @@ export default function PageOptionsPanel() {
...
+                  Only available in paid plans. Click to learn more.
+                </Typography>
+              }
+            >
+              <Chip
+                variant="outlined"
+                color="primary"
+                component={'a'}
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx:138]: I wish prettier would fix these

```suggestion
                component="a"
```

File: packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx [hunk 3]
```diff
@@ -122,7 +122,38 @@ export default function PageOptionsPanel() {
...
+                href="https://mui.com/toolpad/studio/concepts/rbac/"
+                target="_blank"
+                rel="noopener"
+                sx={{
+                  height: '1.5rem',
+                  fontSize: 'inherit',
+                  verticalAlign: 'inherit',
+                  mx: '0.2rem',
+                  '& :hover': {
+                    cursor: 'pointer',
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/PageEditor/PageOptionsPanel.tsx:148]:
```suggestion
```
use the `clickable` prop of the `Chip`

File: packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  feature,
+  warning,
+  variant,
+  action,
+  sx,
+}: {
+  type?: AlertColor;
+  feature?: string;
+  warning?: string;
+  variant?: 'alert' | 'chip';
```
Comment (Janpot) [inline packages/toolpad-studio/src/toolpad/AppEditor/UpgradeNotification.tsx:21]: Just export two components, an alert and a chip

**Review by Janpot — APPROVED** [review]

**Pull request merged**